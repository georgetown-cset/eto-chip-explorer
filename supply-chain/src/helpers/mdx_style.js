import React, {useEffect, memo} from "react";
import Typography from "@mui/material/Typography"

const mdxComponents = {
  p: (() => {
    return memo(props =>
      <Typography {...props} component={"p"} variant={"body1"} style={{marginBottom: "20px"}}/>);
  })(),
  h4: (() => {
    return memo(props =>
      <Typography {...props} component={"h4"} variant={"h4"} style={{margin: "20px 0px 20px 0px"}}/>
    );
  })(),
  h5: (() => {
    return memo(props =>
      <Typography {...props} component={"h5"} variant={"h5"} style={{margin: "20px 0px 10px 0px"}}/>
    );
  })(),
  h6: (() => {
    return memo(props =>
      <Typography {...props} component={"h6"} variant={"h6"} style={{margin: "10px 0px 10px 0px"}}/>);
  })(),
  li: (() => {
    return memo(props =>
      <Typography {...props} component={"li"}/>);
  })(),
  hr: (() => {
    return memo(props =>
      <hr style={{margin: "50px 0px 30px 0px"}}/>);
  })(),
};

export default mdxComponents;
