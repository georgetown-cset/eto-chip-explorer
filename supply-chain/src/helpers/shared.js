import React from "react";
import SvgIcon from '@mui/material/SvgIcon';

const CogIcon = (props) => {
  return (
    <SvgIcon {...props} style={{fillRule: "evenodd", margin: "-8px 0px"}}>
      <svg id="a" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 44 44">
      <path class="b" d="M13.75,29.7c-.2,.11-.46,.04-.57-.16l-2.12-3.68c-.11-.2-.05-.46,.15-.58l1.88-1.15c.2-.12,.36-.4,.36-.64v-3c0-.23-.16-.52-.36-.64l-1.88-1.15c-.2-.12-.26-.38-.15-.58l2.12-3.68c.11-.2,.38-.27,.57-.16l1.94,1.05c.2,.11,.53,.11,.73,0l2.6-1.5c.2-.12,.37-.4,.37-.63l.06-2.21c0-.23,.2-.42,.43-.42h4.24c.23,0,.42,.19,.43,.41l.06,2.21c0,.23,.17,.51,.37,.63l2.6,1.5c.2,.12,.53,.12,.73,0l1.94-1.06c.2-.11,.46-.04,.57,.16l2.12,3.68c.11,.2,.05,.46-.15,.58l-1.88,1.15c-.2,.12-.36,.4-.36,.64v3c0,.23,.16,.52,.36,.64l1.88,1.15c.2,.12,.26,.38,.15,.58l-2.12,3.68c-.11,.2-.38,.27-.57,.16l-1.94-1.05c-.2-.11-.53-.11-.73,0l-2.6,1.5c-.2,.12-.37,.4-.37,.63l-.06,2.21c0,.23-.2,.42-.43,.42h-4.24c-.23,0-.42-.19-.43-.41l-.06-2.21c0-.23-.17-.51-.37-.63l-2.6-1.5c-.2-.12-.53-.12-.73,0l-1.94,1.06Zm5.74-3.37c2.4,1.38,5.46,.56,6.84-1.83,1.38-2.4,.56-5.46-1.83-6.84-2.4-1.38-5.46-.56-6.84,1.83-1.38,2.4-.56,5.46,1.83,6.84Z"/>
      </svg>
    </SvgIcon>
  );
}

const BeakerIcon = (props) => {
  return (
    <SvgIcon {...props} style={{margin: "-8px 0px"}}>
      <svg id="a" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 44 44">
        <path class="b" d="M31.94,10.08H15.23c-1.02,0-2.43,.23-3.39,1.32-.62,.7-.84,1.54-.84,2.09,0,.21,.07,.4,.21,.56,.13,.15,.32,.26,.52,.28,1.58,.22,1.81,.32,1.81,1.28v14.04c0,1.13,.45,2.21,1.24,3.01,.79,.8,1.87,1.25,2.99,1.25h10.15c1.12,0,2.2-.45,2.99-1.25,.79-.8,1.24-1.88,1.24-3.01V13.49c0-.67,.39-1.45,.53-1.67h0s.03-.05,.04-.07c.11-.15,.27-.37,.27-.69,0-.57-.48-.98-1.06-.98Zm-19.03,2.72c.28-.47,.91-1.01,2.34-1.01h15.57c-.19,.47-.36,1.07-.36,1.7v3.4H15.23v-1.28c0-1.99-1.13-2.56-2.32-2.82Z"/>
      </svg>
    </SvgIcon>
  );
}

const getIcon = (nodeType, style) => {
 return nodeType === "tools" ? <CogIcon style={style}/> : <BeakerIcon style={style}/>;
};

export default getIcon;
