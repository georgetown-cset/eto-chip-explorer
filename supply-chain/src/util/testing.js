import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

export function userEventSetup(jsx, options) {
  return {
    user: userEvent.setup(options),
    ...render(jsx),
  }
}
