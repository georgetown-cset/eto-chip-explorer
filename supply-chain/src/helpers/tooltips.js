import {FILTER_INPUT, FILTER_COUNTRY, FILTER_CONCENTRATION, FILTER_ORG} from "../helpers/shared";

const tooltips = {
  [FILTER_INPUT]: "Use this option to jump to a specified stage, process, tool, or material in the supply chain.",
  [FILTER_COUNTRY]: "Highlights chip inputs supplied by a specified country or set of countries.",
  [FILTER_ORG]: "Highlights chip inputs supplied by a specified company or set of companies.",
  [FILTER_CONCENTRATION]: "Highlights chip inputs supplied by a small number of countries.",

  providers: {
    countries: "Countries with significant global market share.",
    orgs: "Global companies with significant market share or otherwise notable capabilities.",
  },
};

export default tooltips;
