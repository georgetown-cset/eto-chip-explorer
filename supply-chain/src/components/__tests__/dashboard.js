import React from "react"
import { render, screen, fireEvent } from "@testing-library/react"
import { userEventSetup } from '../../util/testing';

import Dashboard, {
  GradientLegend
} from "../dashboard";
import {FILTER_INPUT, FILTER_COUNTRY} from "../../helpers/shared";

window.URL.createObjectURL = jest.fn();

describe("Gradient Legend", () => {
  it("renders empty correctly", () => {
    const {asFragment} = render(<GradientLegend type={FILTER_INPUT} numSelected={2}/>);
    expect(asFragment()).toMatchSnapshot();

    expect(screen.queryByText(/[a-z]/).textContent).toEqual("Legend:");
  });

  it("renders with boxes correctly", () => {
    const {asFragment} = render(<GradientLegend type={FILTER_COUNTRY} numSelected={2}/>);
    expect(asFragment()).toMatchSnapshot();

    expect(screen.queryByText("countries have", {exact: false})).not.toBeNull();
  });
});

describe("Dashboard", () => {
  beforeEach(() => {
    window.HTMLElement.prototype.scrollIntoView = function() {};
    const location = {
      ...window.location,
      search: '',
    };
    Object.defineProperty(window, 'location', {
      writable: true,
      value: location,
    });
  })

  it.skip("renders correctly", () => {
    const {asFragment} = render(<Dashboard/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("changes the filter shown", () => {
    render(<Dashboard/>);
    // Initially, only the top-level dropdown is shown
    expect(screen.queryByText("Highlight by", {exact: false})).not.toBeNull();
    expect(screen.queryByText("Choose supplier countries", {exact: false})).toBeNull();

    // Change top level dropdown's value
    fireEvent.mouseDown(screen.getByText("None"));
    fireEvent.click(screen.getByText("Supplier countries"));

    /// Check that the second dropdown is now shown
    expect(screen.queryByText("Highlight by", {exact: false})).not.toBeNull();
    expect(screen.queryByText("Choose supplier countries", {exact: false})).not.toBeNull();

    // Reset dropdown -- not sure why I need to do this, but without this, the
    // changes persist into the next test.
    fireEvent.mouseDown(screen.getAllByText("Supplier countries")[0]);
    fireEvent.click(screen.getByText("None"));
  });

  it("opens the documentation node", async () => {
    const { user } = userEventSetup(<Dashboard />);

    // Click on an input node to show its documentation
    expect(screen.getAllByText("Crystal growing furnaces").length).toEqual(1);
    await user.click(screen.getByText("Crystal growing furnaces"));
    expect(screen.getAllByText("Crystal growing furnaces").length).toEqual(3);

    // Click on another node inside the documentation panel
    await user.click(screen.getAllByText("Deposition")[1]);
    expect(screen.queryAllByText("Some text about N35")).not.toBeNull();

    // Close the documentation panel
    await user.click(screen.getByText("Clear Highlights"));
    expect(screen.getAllByText("Crystal growing furnaces").length).toEqual(1);
  });

  it("opens a documentation node with variants and subvariants", () => {
    render(<Dashboard/>);
    // Click on a node with subvariants
    fireEvent.click(screen.getByText("Deposition tools"));
    fireEvent.click(screen.getAllByText("Chemical vapor deposition tools")[0]);

    // Show subvariants list
    expect(screen.getAllByText("Atomic layer deposition tools").length).toEqual(5);
  });

  it("opens a stage node", () => {
    render(<Dashboard/>);
    // Click on a stage node title
    fireEvent.click(screen.getByText("Fabrication"));

    // Show stage node text
    expect(screen.getByText("Chart shows share of global fabrication capacity", {exact: false})).not.toBeNull();
  });

  it("changes the highlighting shown", () => {
    render(<Dashboard/>);
    // Initially, the nodes are not highlighted
    const inputNode = screen.getByText("Crystal growing furnaces");
    expect(inputNode.parentElement.parentElement.classList).not.toContain("highlighted");
    const stageNode = screen.getByText("Assembly, testing, and packaging (ATP)");
    expect(stageNode.parentElement.classList).not.toContain("highlighted");

    // Change top level dropdown's value
    fireEvent.mouseDown(screen.getByText("None"));
    fireEvent.click(screen.getByText("Market concentration"));

    // Check that the node is now highlighted correctly
    expect(inputNode.parentElement.parentElement.classList).toContain("highlighted");
    expect(inputNode.parentElement.parentElement.classList).toContain("gradient-100");
    expect(stageNode.parentElement.classList).toContain("not-applicable");
  });

});
