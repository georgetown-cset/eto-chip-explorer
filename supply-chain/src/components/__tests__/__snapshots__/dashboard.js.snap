// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gradient Legend renders correctly 1`] = `
<DocumentFragment>
  <div
    class="gradient-legend"
  />
</DocumentFragment>
`;

exports[`Gradient Legend renders empty correctly 1`] = `
<DocumentFragment>
  <div
    class="gradient-legend"
  />
</DocumentFragment>
`;

exports[`Gradient Legend renders with boxes correctly 1`] = `
<DocumentFragment>
  <div
    class="gradient-legend"
  >
    selected countries have 0% market share
    <div
      class="gradient-box-wrapper"
    >
      <div
        class="gradient-box gradient-20"
      />
      <div
        class="gradient-box gradient-40"
      />
      <div
        class="gradient-box gradient-60"
      />
      <div
        class="gradient-box gradient-80"
      />
      <div
        class="gradient-box gradient-100"
      />
    </div>
    selected countries have &gt;80% market share
  </div>
</DocumentFragment>
`;
